src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+  const Real one = 1;
+  const Real three = 3;
+  // BAD CONSTANT
+  const Real dlndg_nimptblgrow = haero::log(1.25);
+  // ! set up temperature-pressure pair to compute impaction scavenging rates
+  const Real temp_0C = haero::Constants::melting_pt_h2o; //     ! K
+  // BAD CONSTANT
+  const Real press_750hPa = 0.75e6; //  ! dynes/cm2
+  for (int imode = 0; imode < AeroConfig::num_modes(); ++imode) {
+    const Real sigmag = sigmag_amode[imode];
+    // Note: we replaced lspectype_amode and lspectype_amode for
+    // dry_aero_density const int ll = lspectype_amode[0][imode]; const Real
+    // rhodryaero = specdens_amode[ll];
+    const Real rhodryaero = aerosol_dry_density[imode];
+    for (int jgrow = nimptblgrow_mind; jgrow <= nimptblgrow_maxd; ++jgrow) {
+      // ratio of diameter for wet/dry aerosols [fraction]
+      const Real wetdiaratio = haero::exp(Real(jgrow) * dlndg_nimptblgrow);
+      // aerosol diameter [m]
+      const Real dg0 = dgnum_amode[imode] * wetdiaratio;
+      // ratio of volume for wet/dry aerosols [fraction]
+      const Real wetvolratio =
+          haero::exp(Real(jgrow) * dlndg_nimptblgrow * three);
+      // dry and wet aerosol density [kg/m3]
+      Real rhowetaero = one + (rhodryaero - one) / wetvolratio;
+      rhowetaero = haero::min(rhowetaero, rhodryaero);
+      /* FIXME: not sure why wet aerosol density is set as dry aerosol density
+         here ! but the above calculation of rhowetaero is incorrect. ! I think
+         the number 1.0_r8 should be 1000._r8 as the unit is kg/m3 ! the above
+         calculation gives wet aerosol density very small number (a few kg/m3) !
+         this may cause some problem. I guess this is the reason of using dry
+         density. ! should be better if fix the wet density bug and use it. Keep
+         it for now for BFB testing ! -- (commented by Shuaiqi Tang when
+         refactoring for MAM4xx) */
+
+      rhowetaero = rhodryaero;
+      /*compute impaction scavenging rates at 1 temp-press pair and save
+              ! note that the subroutine calc_1_impact_rate uses CGS units */
+      // aerosol diameter in CGS unit [cm]
+      const Real dg0_cgs = dg0 * 1.0e2; //  ! m to cm
+      // wet aerosol density in CGS unit [g/cm3]
+      const Real rhowetaero_cgs = rhowetaero * 1.0e-3; //   ! kg/m3 to g/cm3
+      // scavenging rate of aerosol number [1/s]
+      Real scavratenum = zero;
+      // scavenging rate of aerosol volume [1/s]
+      Real scavratevol = zero;
+
+      calc_1_impact_rate(dg0_cgs, sigmag, rhowetaero_cgs, temp_0C, press_750hPa,
+                         scavratenum, scavratevol);
+
+      scavimptblnum[jgrow - nimptblgrow_mind][imode] = haero::log(scavratenum);
+      scavimptblvol[jgrow - nimptblgrow_mind][imode] = haero::log(scavratevol);
+
+    } // jgrow
+
+  } // end imode
+
+} // modal_aero_bcscavcoef_init
+
+} // end namespace aero_model
+
+/// @class aero_model
+/// This class implements MAM4's aero_model parameterization.
```

Comment:
Is `aero_model` a parameterization, or is it a collection of parameterizations like the microphysics stuff that James is working on? If the latter, we might not need a class, but only a top-level function that gets called (eventually).

Reply:
In aero_model, we have a subroutine that invokes microphysics.  Thus, I believe. Yes, it is a collection of parameterizations.  I will delete the class.